### ISSUE TITLE
LC ITS

### ISSUE THREAD

**Issue by larsbrinkhoff** [PDP-10/its#1165]
LC ITS is to be a new KS10 configuration:

- RH11 + eight RP06s.
- 512K memory.
- TM03 + one TU45.
- IMP?
- DZ11?
- Chaosnet?
- Ethernet? (Depends on @atsampson.)

CC @rmaldersoniii.

**Comment by larsbrinkhoff** [comment]
Is the Ethernet card an Interlan NI1010?  We may be able to get that to work, but probably at a later time.

**Comment by atsampson** [comment]
Doesn't that machine have an ITS name (and CONFIG entry) already? :-)

**Comment by rmaldersoniii** [comment]
Of course the machine had a name, but that one is well known and has built-in expectations. We want to start fresh, with more disks hosting more software and the possibility of allowing public accounts.
I thought I had sent the LC config to Lars several months ago. If not, I apologize; I last touched the build in January 2018, then got buried in a different OS and architecture altogether.

**Comment by larsbrinkhoff** [comment]
The microcode RAM file is built fresh from ITS microcode sources.  It's not binary identical to the one that came from here: http://www.cosmic.com/u/mirian/its/files/minsys.tape

@blipvert, do you know where the .; RAM RAM file on the minsys tape came from?

**Comment by larsbrinkhoff** [comment]
I have eight RP06 images built for LC ITS.  I used the KLH10 emulator for this, since it supports Chaosnet.  Unfortunately, it only has a stub device for DZ11 so I haven't tested that.

I haven't considered the file format.  I believe the disk emulator uses the SIMH format, so I think I need to rebuild with KLH10 using that.  Then I can cross check the images with SIMH, but I'm quite sure it'll complain about CH11 missing.

**Comment by larsbrinkhoff** [comment]
It just occurred to me, we should probably increase NUDSL to have more user directories available.  The current value of 500. is large for an ITS, but I guess LC will be even larger in terms of disk space.

**Comment by rmaldersoniii** [comment]
Yes, the MDE uses SimH format for disk images, but KLH10 is agnostic about disk images. If instead of declaring a disk "DBD9", you declare it "DLW8", you get a SimH-compatible disk image and don't have to worry about CH11 not existing on the SimH KS10 emulation.

**Comment by larsbrinkhoff** [comment]
I'm using DLW8 now, and verified that SIMH likes this format.  The disk image will boot to DSKDMP.

I see the image files vary in size.  Does MDE need them padded with zeroes?

**Comment by larsbrinkhoff** [comment]
Possible issues:

- Microcode wrong?
- DZ11 boards, are they working?
- Can ITS even be run without DZ11?  KLH10 implies not.

**Comment by larsbrinkhoff** [comment]
I think we can claim the microcode is verified now.  The version running is freshly built from source code.

**Comment by larsbrinkhoff** [comment]
- CH11 board, is it ok?
- NI1010 board.  ITS support hasn't been finished.  But if it has been sitting there all along, maybe there's no harm in leaving it there.

**Comment by rmaldersoniii** [comment]
Because LC uses an LCM+L MDE rather than DEC spinning rust, it may be that the KS can handle RP07 images (where real RP07s are too fast for the KS-10's Unibus). At some point we should test this out.

**Comment by larsbrinkhoff** [comment]
I understand the MDE doesn't emulate a tape drive.  That is unfortunate, because backing up to tape is a convenient way to move files from one system to another.  Or the same system with a different disk configuration.

**Comment by rmaldersoniii** [comment]
There is a variant, the MTE, which works as 8 TM78 controllers with a single drive each (because subunits are difficult to implement in the Xilinx). The KS-10 does not support the TM78, only the TM03/TM02 style controller. One of the engineers is working on making the MTE talk TM03, but it's not high on his priority list.

**Comment by larsbrinkhoff** [comment]
I believe RP07 support was added at the very last minute and never tested at MIT and hence quite possibly never used on a real KS10 running ITS.  I don't know if anyone ran it with KLH10 or SIMH; SV and UP both use RP06.

**Comment by rmaldersoniii** [comment]
I used an RP07 early in my experiments with your Github project, under KLH10 and SimH. It worked OK as I recall for a single-disk system, but multiple RP07s got into trouble very quickly. Since this was not a priority at the museum, I did not look into what was actually failing. Seems like something for a younger hacker to think about.

**Comment by larsbrinkhoff** [comment]
I'm breeding a young hacker, but he's only five.
